lib/ansible/cli/pull.py

```diff
@@ -123,8 +125,12 @@ def parse(self):
                                help='modified files in the working repository will be discarded')
         self.parser.add_option('--track-subs', dest='tracksubs', default=False, action='store_true',
                                help='submodules will track the latest changes. This is equivalent to specifying the --remote flag to git submodule update')
+        # add a subset of the check_opts flag group manually, as the full set's
+        # shortcodes conflict with above --checkout/-C
         self.parser.add_option("--check", default=False, dest='check', action='store_true',
                                help="don't make any changes; instead, try to predict some of the changes that may occur")
+        self.parser.add_option("--diff", default=C.DIFF_ALWAYS, dest='diff', action='store_true',
+                               help="when changing (small) files and templates, show the differences in those files; works great with --check")
```

Comment:
tempted to say just use check_opts=True and remove syntax check afterward with remove_option (we do this in ansible-inventory).

Reply:
If we did that, we'd have to remove `--check` too and and re-add it without it's shorthand `-C`? Instead of `check_opts=False` with two adds we'd end up with a `True` and two removals and one re-add. I suppose it would reduce the number of duplicate help strings by one. You prefer the second option? Then I'll invert the current PR to do it that way.

Reply:
if it were not for the `-C` i would not say 'tempted' but actually request it

Reply:
Okay, I'll consider this merge-ready then.